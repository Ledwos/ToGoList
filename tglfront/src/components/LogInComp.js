import React from 'react';

const LogInComp = () => {
    return (
        <div>
            <p>I'm the log in component</p>
        </div>
    );
};

export default LogInComp;