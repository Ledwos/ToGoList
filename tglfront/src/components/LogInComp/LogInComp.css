#loginDiv {
    margin-top: 70px;
    width: 100%;
    text-align: center;
}

#loginForm {
    /* border: 2px solid red; */
    display: inline-flex;
    flex-direction: column;
    width: 60%;
    max-width: 300px;
    justify-self: center;
}

#u_email, #u_pass {
    margin: 10px;
    padding: 10px;
    font-size: 16px;
    border: 3px solid rgb(61, 178, 255);
    border-radius: 3px;
}

#loginBtn {
    border: 3px solid rgb(61, 178, 255);
    border-radius: 3px;
    width: 50%;
    padding: 10px 0px;
    margin-top: 20px;
    align-self: center;
    box-shadow: 3px 3px 0px 0px rgb(44, 127, 182);
    background-color: rgb(204, 235, 255);
}

#loginBtn:hover {
    cursor: pointer;
    background-color: rgb(61, 178, 255);
    color: white;
}

#loginBtn:active {
    box-shadow: 0px 0px 0px 0px rgb(44, 127, 182);
    margin: 23px -3px -3px 3px;
}

#clickHere {
    color: gray;
}

#clickHere:hover {
    cursor: pointer;
}

#clickLogin {
    color: gray;
}

#clickLogin:hover {
    cursor: pointer;
}
