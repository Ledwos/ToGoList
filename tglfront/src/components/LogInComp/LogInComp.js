import React from 'react';
import './LogInComp.css';

const LogInComp = (props) => {
    return (
        <div id='loginDiv'>
            <form onSubmit={props.handleLogin} method='POST' id='loginForm'>
                {/* <label for='u_email'>email: </label> */}
                <input 
                    type='text' 
                    id='u_email' 
                    placeholder='someone@organised.com'
                    pattern="[a-zA-Z0-9!#$%&amp;'*+\/=?^_`{|}~.-]+@[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)*"
                    required
                    ></input>
                {/* <label for='u_pass'>password: </label> */}
                <input 
                    type='password' 
                    id='u_pass'
                    placeholder='password'
                    required
                    ></input>
                <button type='submit' id='loginBtn'>Log in</button>
            </form>
            <br />
            <br />
            <br />
            <p>Don't have an account? <span onClick={props.toggleSignup} id='clickHere'>click here</span> to sign up</p>
            {/* <p onClick={props.resetHome}>X</p> */}
        </div>
    );
};

export default LogInComp;