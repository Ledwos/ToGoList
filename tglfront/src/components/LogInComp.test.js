import React from 'react';
import LogInComp from './LogInComp';
import renderer from 'react-test-renderer';

it('renders LogInComp component', () => {
    const LogInComponent = renderer.create(<LogInComp />).toJSON();
    expect(LogInComponent).toMatchSnapshot();
});