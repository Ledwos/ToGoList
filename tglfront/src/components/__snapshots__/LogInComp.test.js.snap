// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LogInComp component 1`] = `
<div>
  <p>
    I'm the log in component
  </p>
</div>
`;
